### DISCUSSION TITLE
command aliases?

### DISCUSSION THREAD

**Discussion by nschloe** [ewels/rich-click#91] [Q&A]
There are diffent possibilities for getting command aliases (e.g., `git ci` for `git commit`) in click. See [here](https://click.palletsprojects.com/en/8.1.x/advanced/) or [here](https://github.com/click-contrib/click-aliases). Unfortunately, none of these plays nicely with rich-click in that formatting is removed.

Any hints here?

**Comment by ewels** [comment]
Oh nice! Sorry, I haven't used this myself so it likely just needs to be added. PRs welcome for this 😊

**Comment by MoffysDev** [comment]
You can combine click‑aliases with rich‑click by making the alias group inherit from both `rich_click.RichGroup` (or `RichCommand`) and `click_aliases.AliasGroup`. 
rich‑click only looks for a subclass of its own rich group/command to apply styling, so as long as your class is a `rich_click.RichGroup` the formatting will be preserved.

```python
import rich_click as click
from click_aliases import AliasGroup

class AliasedRichGroup(rich_click.RichGroup, AliasGroup):
 pass

@click.command(cls=AliasedRichGroup)
def cli():
 pass

@cli.command()
def commit():
 """Alias: ci"""
 click.echo("committing")

# Now `git ci` (or `yourtool ci`) works and the help output stays rich‑styled.
```

If you prefer to use the built‑in alias
